src/EditorFeatures/Core/Implementation/CommentSelection/CommentSelectionResult.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.CommentSelection
+{
+    internal struct CommentSelectionResult
+    {
+        /// <summary>
+        /// Text changes to make for this operation.
+        /// </summary>
+        public ImmutableList<TextChange> TextChanges { get; }
+        /// <summary>
+        /// Tracking spans used to format and set the output selection after edits.
+        /// </summary>
+        public ImmutableList<CommentTrackingSpan> TrackingSpans { get; }
```

Comment:
Lists/ImmutableLists are uncommon in the IDE now.  Did you need to use these for some specific purpose?  or would ImmutableArray be available?

Reply:
Nope, immutable array would work.  Didn't know there was a preference.  Will change.